pkg/driver/driver_aws.go

```diff
@@ -142,6 +117,77 @@ func (d *AWSDriver) Create() (string, string, error) {
 	return d.encodeMachineID(d.AWSMachineClass.Spec.Region, *runResult.Instances[0].InstanceId), *runResult.Instances[0].PrivateDnsName, nil
 }
 
+func (d *AWSDriver) generateSecurityGroups(sgIDs []string) ([]*string, error) {
+
+	var securityGroupIDs []*string
+	for _, sg := range sgIDs {
+		securityGroupIDs = append(securityGroupIDs, aws.String(sg))
+	}
+	return securityGroupIDs, nil
+}
+
+func (d *AWSDriver) generateTags(tags map[string]string) (*ec2.TagSpecification, error) {
+
+	// Add tags to the created machine
+	tagList := []*ec2.Tag{}
+	for idx, element := range tags {
+		if idx == "Name" {
+			// Name tag cannot be set, as its used to identify backing machine object
+			continue
+		}
+		newTag := ec2.Tag{
+			Key:   aws.String(idx),
+			Value: aws.String(element),
+		}
+		tagList = append(tagList, &newTag)
+	}
+	nameTag := ec2.Tag{
+		Key:   aws.String("Name"),
+		Value: aws.String(d.MachineName),
+	}
+	tagList = append(tagList, &nameTag)
+
+	tagInstance := &ec2.TagSpecification{
+		ResourceType: aws.String("instance"),
+		Tags:         tagList,
+	}
+	return tagInstance, nil
+}
+
+func (d *AWSDriver) generateBlockDevices(blockDevices []v1alpha1.AWSBlockDeviceMappingSpec, rootDeviceName *string) ([]*ec2.BlockDeviceMapping, error) {
+
+	var blkDeviceMappings []*ec2.BlockDeviceMapping
+	// if blockDevices is empty, AWS will automatically create a root partition
+	for _, disk := range blockDevices {
+
+		deviceName := disk.DeviceName
+		if disk.DeviceName == "/root" || len(blockDevices) == 1 {
+			deviceName = *rootDeviceName
+		}
+		volumeSize := disk.Ebs.VolumeSize
+		volumeType := disk.Ebs.VolumeType
+		deleteOnTermination := disk.Ebs.DeleteOnTermination
+		iops := disk.Ebs.Iops
```

Comment:
This variable is not used anywhere. Shouldn't the build fail here?

Reply:
Yes just pushed the fix :)